doc/influxql-2.md

```diff
@@ -0,0 +1,910 @@
+// Returns: matrix of uint64 values or other value converted to a uint64
+```
+
+##### String
+`string` will convert a single value or all values in a matrix to `string` type values.
+
+```javascript
+// Accepts: matrix, float64, int64, uint64, bool
+// Arguments:
+//   error - boolean indicating if it should fail on casting error. default is false
+// Returns: matrix of string values or other value converted to a string
+```
+
+##### Bool
+`bool` will convert a single value or all values in a matrix to `bool` type values.
+
+```javascript
+// Accepts: matrix, float64, int64, uint64, string
+// Arguments:
+//   error - boolean indicating if it should fail on casting error. default is false
+// Returns: matrix of bool values or other value converted to a bool
+```
+
+#### Window
+`window` will group points is each series in a matrix based on the windowing criteria. These windows can then have functions applied to them to produce aggregates or summaries of each.
+
+```javascript
+// Accepts: a matrix
+// Arguments:
+//   every - start a window after this duration. Must be less than or equal to period.
+//   period - windows are for this duration. If only every or period is specified,
+//            the other is set to the same value by default. If the every duration
+//            is less than the period, windows will overlap.
+//   count - number of windows (points) desired. Will adjust bucket durations
+//           based on this count and the start and end time. Either count or
+//           every & period can be specified, but not both.
+//   start - optional, time to start the windows. If not specified, the time to start
+//           will be the time of the first value in each series
+//   round - optional, duration to round the start time to
+// Returns: matrix
+```
+
+For example, we start with this matrix `foo`:
+
+|                           Series | 10  | 20  | 30  | 40
+|                              --- |:---:|:---:|:---:|:---:
+|`{"host":"a","metric":"cpu_user"}`| 12  | 13  | 18  | 14
+|`{"host":"b","metric":"cpu_user"}`| 6   | 5   | 3   | 2
+
+And we call `foo.window(every:20s)` we are conceptually splitting the series into windows demarcated every 20 seconds. For the aid of this visualization we have empty columns to indicate the start and end of a window:
+
+|                           Series |1.start|  10 |  20 |1.end|2.start| 30  | 40  | 2.end
+|                              --- |-------|:---:|:---:|-----|-------|:---:|:---:|---
+|`{"host":"a","metric":"cpu_user"}`|       | 12  | 13  |     |       | 18  | 14  |
+|`{"host":"b","metric":"cpu_user"}`|       | 6   | 5   |     |       | 3   | 2   |
+
+We see that there are now columns that mark the start and end of each window. This is meant as a temporary stage before calling some sort of aggregate or selector function to be applied to each cell. For example if we had called `foo.window(every:20s).sum()` we would have received the following result:
+
+Series | 10  | 30
```

Comment:
Do we want the ability to change the behavior of all aggregate functions via an options like syntax?
Meaning it can be cumbersome and error prone to have to specify and optional argument to all aggregation functions because you need to use the min time bound.

Maybe something like this at the top of a script to set defaults for just that script:

```
option time_bound=min
```

This has uses beyond just the time bound. For example the user may want to specify how out of order data is handled. The options are to drop it, wait for it for X amount of time (which delays the result), or recompute results when old data arrives.

This could be set via an options syntax.

```
option out_of_order=drop
```
 or

```
option out_of_order=wait 5m
```

```
option out_of_order=recompute
```

Thoughts?